api/server/app.cc

```diff
@@ -456,34 +458,58 @@ static const char *SrcCgroup() {
     return NULL;
 }
 
+int setCgroups(std::string file1, std::string file2) {
+    std::ifstream ifs(file1);
+    std::ofstream ofs(file2);
+    if ((ifs.is_open()) && (ofs.is_open())) {
+        ofs << ifs.rdbuf();
+        return 0;
+    } else {
+        return -1;
+    }
+}
+
 static int InitCgroup(int multiplier) {
     const char *cgroupPath = SrcCgroup();
-
+    int check = 0;
+ 
     if (!cgroupPath)
         return -1;
-    std::string create_dir("mkdir " + std::string(cgroupPath) + "/butterfly");
-
-    BASH(create_dir) {
-        LOG_WARNING_("can't create butterfly cgroup, fail cmd '%s'",
-                     create_dir.c_str());
+    std::string directory = std::string(cgroupPath) + "/butterfly";
+    if (mkdir(directory.c_str(), S_IWGRP) < 0) {
+        LOG_WARNING_("can't create directory already exists\n");
     }
-    BASH("echo $(( `cat " + std::string(cgroupPath) + "/cpu.shares` * " +
-         std::to_string(multiplier) + " )) > " +
-         cgroupPath + "/butterfly/cpu.shares") {
+ 
+    std::string fileToRead = std::string(cgroupPath) + "/cpu.shares";
+    std::string fileToWrite = std::string(cgroupPath) + "/butterfly/cpu.shares";
+    std::ifstream in(fileToRead);
+    std::ofstream out(fileToWrite);
+    if ((in.is_open()) && (out.is_open())) {
+        std::string s;
+        while (getline(in, s)) {
```

Comment:
a multiplier been a single number in a file, I don't think the loop is necessary